spec/immutable-data-structures.html

```diff
@@ -6,26 +6,521 @@ <h1>Immutable Data Structures</h1>
+        <p>When the `map` method is called with one or two arguments, the following steps are taken:</p>
+        <emu-alg>
+          1. Let _T_ be ? thisTupleValue(*this* value).
+          1. Let _list_ be ? CreateListFromTuple(_T_)
+          1. Let _len_ be the number of elements in _list_.
+          1. If IsCallable(_callbackfn_) is *false*, throw a *TypeError* exception.
+          1. Let _newList_ be a new empty List.
+          1. Let _k_ be 0.
+          1. Repeat, while _k_ &lt; _len_,
+            1. Let _kValue_ be _list_[_k_].
+            1. Let _mappedValue_ be ? Call(_callbackfn_, _thisArg_, &laquo; _kValue_, _k_, _T_ &raquo;).
+            1. Append _mappedValue_ to the end of list _newList_.
+            1. Set _k_ to _k_ + 1.
+          1. Return ? CreateTupleFromList(_newList_).
+        </emu-alg>
+      </emu-clause>
+      <emu-clause id="sec-tuple.prototype.reduceRight">
+        <h1>Tuple.prototype.reduceRight ()</h1>
+      </emu-clause>
+      <emu-clause id="sec-tuple.prototype.some">
+        <h1>Tuple.prototype.some</h1>
+        <p>The initial value of the *Tuple.prototype.some* is the *"some"* property of the intrinsic object %Array.prototype%.</p>
+      </emu-clause>
+      <emu-clause id="sec-tuple.prototype.unshifted">
+        <h1>Tuple.prototype.unshifted ( ..._args_ )</h1>
+        <p>When the *unshifted* method is called, it returns a tuple containing the elements of the tuple, with the arguments to the function invocation prepended.</p>
+        <p>The following steps are taken:</p>
+        <emu-alg>
+          1. Let _T_ be ? thisTupleValue(*this* value).
+          1. Let _argsList_ be a List containing the arguments passed to this function invocation.
+          1. Let _tupleList_ be ? CreateListFromTuple(_T_)
+          1. Let _list_ be a new empty List.
+          1. Let _n_ be 0.
+          1. Repeat, while _argsList_ is not empty,
+            1. If _n_ + 1 &gt; 2<sup>53</sup> - 1, throw a *TypeError* exception.
+            1. Remove the first element from _argsList_, and let _E_ be the value of the element.
+            1. Append _E_ to the end of list _list_.
+            1. Set _n_ to _n_ + 1.
+          1. Repeat, while _tupleList_ is not empty,
+            1. If _n_ + 1 &gt; 2<sup>53</sup> - 1, throw a *TypeError* exception.
+            1. Remove the first element from _tupleList_, and let _E_ be the value of the element.
+            1. Append _E_ to the end of list _list_.
+            1. Set _n_ to _n_ + 1.
+          1. Return ? CreateTupleFromList(_list_).
+        </emu-alg>
+      </emu-clause>
+      <emu-clause id="sec-tuple.prototype.values">
+        <h1>Tuple.prototype.values ()</h1>
+        <p>When the *values* method is called, it returns an iterator over the values of the tuple.</p>
+        <p>The following steps are taken:</p>
+        <emu-alg>
+          1. Let _T_ be ? thisTupleValue(*this* value).
+          1. Return CreateTupleIterator(_T_, ~value~).

```

Comment:
I guess we can reuse Array iterators (here and elsewhere) as TypedArrays do.

Reply:
We could totally reuse Array iterators, I just pushed a commit to add Tuple iterators, but I'm fine with either approach.

Reply:
Given the precedent for TypedArrays reusing the Array iterator but making their own methods, I think you should go with Array iterators here.